app/app/api/api_v1/endpoints/students.py

```diff
@@ -0,0 +1,85 @@
+import logging
+from typing import Any, List
+
+from fastapi import APIRouter, Depends
+from sqlalchemy.orm import Session
+
+from app import crud, models, schemas
+from app.api import deps
+from app.exceptions import ForbiddenException, NotFoundException
+from app.schemas import AdminPermissions
+
+router = APIRouter()
+
+
+@router.get("/", response_model=List[schemas.Student])
+def read_students(
+    db: Session = Depends(deps.get_db),
+    skip: int = 0,
+    limit: int = 100,
+    _: models.Admin = Depends(deps.get_current_active_admin_with_permission("student")),
+) -> Any:
+    """
+    Retrieve students
+    """
+    return crud.student.get_multi(db, skip=skip, limit=limit)
+
+
+@router.get("/me", response_model=schemas.Student)
+def get_student_me(
+    current_student: models.Student = Depends(deps.get_current_student),
+) -> Any:
+    """
+    Get current student
+    """
+    return current_student
+
+
+@router.get("/{student_id}", response_model=schemas.Student)
+def read_student_by_id(
+    student_id: int, current_user: models.User = Depends(deps.get_current_user), db: Session = Depends(deps.get_db)
+) -> Any:
+    """
+    Get a specific student by ID.
+    """
+
+    # Fetch student with the corresponding ID from DB
+    student = crud.student.get(db, id=student_id)
+
+    # Return the fetched object without checking perms if current_student is trying to fetch itself
+    if current_user.id == student_id:
```

Comment:
I was actually wondering.... is there any point in having this if we have a separate `get_student_me`?

Reply:
There is pretty much minimal use for it, but I wanted it in there incase we ever need to fetch a specific's student's details - since that will include the user+term info

Reply:
I mean, we're getting all the info from `get_student_me` as well, just seems unnecessary. There really isn't a usecase when we're gonna use this, since in any case we should be using `get_student_me`. Letting a non admin use `get_student` just seems wrong as well.

@adityavdesai what's your opinion?

Reply:
get_student_me would be for a logged in student, this can be used by admins with student permissions, to retrieve details of a single student

I guess we'll likely have a table of students anyway, but we might have a usecase for this sometime IMO. (get list of all, get yourself, get by id should be available in all APIs I feel)

Reply:
That was my point tbh, if this was far students for admin permissions why are we letting non admins call it? To offer a functionality already offered by another API endpoint?

Reply:
> get_student_me would be for a logged in student, this can be used by admins with student permissions, to retrieve details of a single student
> 
> I guess we'll likely have a table of students anyway, but we might have a usecase for this sometime IMO. (get list of all, get yourself, get by id should be available in all APIs I feel)

Yeah but imo keeping get_by_id() check admin perms pehle hi better (admin only API), not inside here to allow for self. get_student_me() is enough for that

Reply:
Actually, this reminds me, @ksdfg you were the one that implemented this in admin API, @akhilnarang just recreated it here. So this stuff exists in admin too

Reply:
Yeah, majority of my work was based on the admin equivalent.

Reply:
It's actually been bugging me for a while. I get some places where it was required like school, year etc. but even that we get by just querying the user or student object now due to sqlalchemy relationships. Hence wondered if it was OK to rollback this little loophole.